src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -417,6 +417,23 @@ template class LogicalExpression<LogicalOperator::AND>;
 template class LogicalExpression<LogicalOperator::OR>;
 
 namespace detail {
+//______________________________________________________________________________
```

Comment:
This can be a public nonVirtual member function on the `PrefilterExpression` which should simply be called `evaluate`, 
should document the behavior (must be sorted and unique, but hte first and last block may contain wrong elements).

The existing `evaluate` should be made private (possibly with `FRIEND_TEST` for the one check).

Reply:
Actually you can implement this by adding a `bool` flag to the existing evaluate (where the bool means `shall i remove the first/last block and readd, or shall i throw`.  (as an impl).
The recursive calls should have the default argument `always throw` and the public function calls this function with explicit `ytou may remove and readd`.